src/main/kotlin/org/rust/ide/refactoring/RsImportOptimizer.kt

```diff
@@ -125,14 +129,12 @@ class RsImportOptimizer : ImportOptimizer {
                     val useSpeck = it.useItem.useSpeck ?: return@filter false
                     optimizeUseSpeck(psiFactory, useSpeck)
                 }
-                .groupBy({ it.packageGroupLevel }, { it })
-                .map { (groupLevel, useItemWrapper) ->
-                    val useItems = useItemWrapper
-                        .sortedBy { it.useSpeckText }
-                        .mapNotNull { it.useItem.copy() as? RsUseItem }
-                    groupLevel to useItems
-                }
-                .sortedBy { it.first }
+                .sorted()
+                .groupByTo(
+                    sortedMapOf(),
+                    { it.packageGroupLevel },
+                    { it.useItem.copy() as RsUseItem }
+                )
```

Comment:
I'd personally drop `groupByTo` and just track `packageGroupLevel` of a previously inserted item in loop below to add new lines when needed but it's more nitpick because it shouldn't affect performance anyhow